lib/effects/sequence_effect.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flame/components/component.dart';
+import 'package:meta/meta.dart';
+
+import './effects.dart';
+
+class SequenceEffect extends PositionComponentEffect {
+  final List<PositionComponentEffect> effects;
+  int _currentIndex = 0;
+  PositionComponentEffect currentEffect;
+  bool _currentWasAlternating;
+  double _driftModifier = 0.0;
+
+  SequenceEffect({
+    @required this.effects,
+    isInfinite = false,
+    isAlternating = false,
+  }) : super(isInfinite, isAlternating) {
+    /// All effects need to be finite, otherwise the sequence will get stuck
+    assert(effects.every((effect) => !effect.isInfinite));
+
+    /// No effects can be added to components from the start
+    assert(effects.every((effect) => effect.component == null));
+  }
+
+  @override
+  void initialize(PositionComponent _comp) {
+    super.initialize(_comp);
+    _currentIndex = 0;
+    final originalSize = _comp.toSize();
+    final originalPosition = _comp.toPosition();
+    final originalAngle = _comp.angle;
+    effects.forEach((effect) {
+      effect.reset();
+      _comp.setBySize(endSize);
+      _comp.setByPosition(endPosition);
+      _comp.angle = endAngle;
```

Comment:
tbh I'm still a bit puzzled as to why you need to do this. but I don't think it's a merge blocker, after it's merged I can try to take a look to see if I understand it or see a way that doesn't require this.

I think this is at least better than SequenceEffect assuming the implementations of Effect are bounded.

Reply:
Because otherwise you can't calculate the traveltime for each individual effect, and then as a result not get the traveltime of the sequence. Since sometimes depending on where in the chain an effect is it will have different travel time.